### PR TITLE
boards: arm: lpcxpresso55s36: add boot and slot1 flash partitions.

### PR DESCRIPTION
**Pull request by butok**
Add boot_partition and slot1_partition to the lpcxpresso55s36 dts.

### CODE REVIEW

**Comment by butok** [comment]
UP

**Review by iuliana-prodan — APPROVED** [review]

File: boards/arm/lpcxpresso55s36/lpcxpresso55s36.dts [hunk 1]
```diff
@@ -17,7 +17,7 @@
 	chosen {
 		zephyr,sram = &sram0;
 		zephyr,flash = &flash0;
-		zephyr,code-partition = &sramx;
+		zephyr,code-partition = &slot0_partition;
```
Comment (iuliana-prodan) [inline boards/arm/lpcxpresso55s36/lpcxpresso55s36.dts:20]: Please fix typo in commit message: s/slot1_aprtition/slot1_partition

Comment (butok) [reply boards/arm/lpcxpresso55s36/lpcxpresso55s36.dts:20]: > Please fix typo in commit message: s/slot1_aprtition/slot1_partition

Fixed. Thank you.

**Comment by butok** [comment]
UP

File: boards/arm/lpcxpresso55s36/lpcxpresso55s36.dts [hunk 2]
```diff
@@ -143,20 +143,33 @@
 	pinctrl-names = "default";
 };
 
+/* Flash is divided into 32 kB sub-regions.
+ * Each sub-region can be assigned individual
+ * security tier in secure AHB controller.
```
Comment (decsny) [inline boards/arm/lpcxpresso55s36/lpcxpresso55s36.dts:148]: just curious, do you have an opinion on how a zephyr user would interact with the AHB controller?

**Review by decsny — APPROVED** [review]

File: boards/arm/lpcxpresso55s36/lpcxpresso55s36.dts (hunk 2 above)
Comment (butok) [reply boards/arm/lpcxpresso55s36/lpcxpresso55s36.dts:148]: > just curious, do you have an opinion on how a zephyr user would interact with the AHB controller?

There is no any option. MCUx SDK does not have AHB driver.

**Pull request merged**